webapp/channels/src/components/post/post_options.tsx

```diff
@@ -123,6 +124,8 @@ const PostOptions = (props: Props): JSX.Element => {
         );
     }
 
+    const sidebarDimensions = useFollowElementDimensions('sidebar-right');
```

Comment:
i have doubts regarding this approach because 
- this component gets re-rendered multiple times on hover over a post
- although this checks for sidebar-right, this hooks is also called irrespective if this component was in center or in rhs and yes hooks cannot be conditional.

I am wondering if there are other ways to see if something more static. 
cc @calebroseland any thoughts

Reply:
@M-ZubairAhmed @TheRealJoeFriel Yeah, that's a lot of extra `ResizeObserver`s (every post) when we only care about the currently-open post menu.

I think it be easy enough to use CSS. Support for [Container Queries](https://developer.mozilla.org/en-US/docs/Web/CSS/@container#browser_compatibility) and [nth-child of selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child#the_of_selector_syntax) is pretty good now—latest version of all mainstream browsers is enough for this particular use case. (And if the client/browser doesn't support either, it's not catastrophic to display all 3.)

You'd only need to add a class (e.g. `post-menu__emoticon--wrapper`) to the emoji wrapper div, and hardcode the count to `size={3}`, along with this CSS to make it responsive as you currently have it:

```css
#sidebar-right {container: sidebar-right / inline-size;}

@container sidebar-right (width < 500px) {
    .post-menu > div:nth-child(n+2 of .post-menu__emoticon--wrapper) {
        display: none;
    }
}
```

Reply:
@calebroseland thank you for your input. @TheRealJoeFriel let us know if you need help in implementing above

Reply:
Thank you both for the comments, I will look into the proposed solution. I have a busy next few weeks but I will try to find time to work this in ASAP!

Reply:
Just wanted to circle back with a non-css option after a second glance. Since the menu is dependent on `hoverLocal` which causes a rerender, you could query `#sidebar-right`'s width without the reactive part.

```tsx
const showMoreReactions =
            props.location !== Locations.CENTER &&
            (document.getElementById('sidebar-right')?.getBoundingClientRect()?.width ?? 0) > 400;
```

That would have fewer CSS complexities and keep the conditional logic in one place.